flocker/apiclient/_client.py

```diff
@@ -446,8 +514,13 @@ def _request(self, method, path, body, success_codes, error_codes=None):
         :param set success_codes: Expected success response codes.
         :param error_codes: Mapping from HTTP response code to exception to be
             raised if it is present, or ``None`` to set no errors.
+        :param with_headers: Boolean indicating whether headers are required.
+        :param configuration_tag: If not ``None``, include value as
+            ``X-If-Configuration-Matches`` header.
 
-        :return: ``Deferred`` firing with decoded JSON.
+        :return: ``Deferred`` firing with decoded JSON, or if
```

Comment:
:grey_question: This API looks a little gross, but seems to be internal only so probably okay as is.

Slight preference for making a new method `_request_with_headers` that has the body of this function and with_headers set to True. Then changing the content of this function to be a call to `d = self._request_with_headers(...); d.addCallback(lambda x: x[0]); return d` or equivalent.